scripts/run-clang-tidy.py

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/env python3
+# TheSuperHackers @build JohnsterID 15/09/2025 Add clang-tidy runner script for code quality analysis
+# TheSuperHackers @build bobtista 04/12/2025 Simplify script for PCH-free analysis builds
+
+"""
+Clang-tidy runner script for GeneralsGameCode project.
+
+This is a convenience wrapper that:
+- Auto-detects the clang-tidy analysis build (build/clang-tidy)
+- Filters source files by include/exclude patterns
+- Processes files in batches to handle Windows command-line limits
+- Provides progress reporting
+
+For the analysis build to work correctly, it must be built WITHOUT precompiled headers.
+Run this first:
+  cmake -B build/clang-tidy -DCMAKE_DISABLE_PRECOMPILE_HEADERS=ON -G Ninja
+"""
+
+import argparse
+import json
+import multiprocessing
+import subprocess
+import sys
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+
+def find_clang_tidy() -> str:
+    """Find clang-tidy executable in PATH."""
+    try:
+        result = subprocess.run(
+            ['clang-tidy', '--version'],
+            capture_output=True,
+            text=True,
+            timeout=5
+        )
+        if result.returncode == 0:
+            return 'clang-tidy'
+    except (FileNotFoundError, subprocess.TimeoutExpired):
+        pass
+    
+    raise RuntimeError(
+        "clang-tidy not found in PATH. Please install clang-tidy:\n"
+        "  Windows: Install LLVM from https://llvm.org/builds/"
+    )
+
+
+def find_project_root() -> Path:
+    """Find the project root directory."""
+    current = Path(__file__).resolve().parent
+    while current != current.parent:
+        if (current / 'CMakeLists.txt').exists():
+            return current
+        current = current.parent
+    raise RuntimeError("Could not find project root (no CMakeLists.txt found)")
+
+
+def find_compile_commands(build_dir: Optional[Path] = None) -> Path:
+    """Find compile_commands.json from the clang-tidy analysis build."""
+    project_root = find_project_root()
+
+    if build_dir:
+        if not build_dir.is_absolute():
+            build_dir = project_root / build_dir
+        compile_commands = build_dir / "compile_commands.json"
+        if compile_commands.exists():
+            return compile_commands
+        raise FileNotFoundError(
+            f"compile_commands.json not found in {build_dir}"
+        )
+
+    # Use the dedicated clang-tidy build (PCH-free, required for correct analysis)
+    clang_tidy_build = project_root / "build" / "clang-tidy"
+    compile_commands = clang_tidy_build / "compile_commands.json"
+
+    if not compile_commands.exists():
+        raise RuntimeError(
+            "Clang-tidy build not found!\n\n"
```

Comment:
This error is misleading as its due to the missing compile-commands.json, not the missing build folder. Help text should suggest a cmake command that includes `-DCMAKE_EXPORT_COMPILE_COMMANDS=ON`